mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td

```diff
@@ -518,8 +526,8 @@ def StoreTileSliceOp : ArmSME_Op<"store_tile_slice"> {
   }];
 
   let assemblyFormat = [{
-    $tile `,` $tile_slice_index `,` $base `[` $indices `]` (`layout` `` $layout^)?
-      attr-dict `:` type($base) `,` type($tile)
+    $tile `,` $tile_slice_index `,` $mask `,` $base `[` $indices `]` (`layout` `` $layout^)?
+      attr-dict `:` type($base) `,` type($mask) `,` type($tile)
```

Comment:
Is this `type($mask)` needed? It should be inferred from your `TileSliceMaskConstraint`

Reply:
No it can be inferred as you say, but I decided to add the type anyway since this maps 1-1 with intrinsic. `tile_load`/`tile_store` are aligned with Vector transfer read/write so I didn't add mask type to asm for those.

The type is also present in `load_tile_slice`, we could remove in a follow-up if desired? I've don't mind either way.

Reply:
I prefer omitting the types that are very clear from context (i.e. the mask size must match the slice), but this is fine for now.

Reply:
Ok cool cheers, I'll post a follow-up :)